components/automate-cli/pkg/verifyserver/services/batchcheckservice/trigger/certificatechecktrigger/certificatechecktrigger.go

```diff
@@ -25,9 +25,14 @@ func NewCertificateCheck(log logger.Logger, port string) *CertificateCheck {
 	}
 }
 
-func (ss *CertificateCheck) Run(config models.Config) []models.CheckTriggerResponse {
+func (ss *CertificateCheck) Run(config *models.Config) []models.CheckTriggerResponse {
 	ss.log.Info("Performing Certificate check from batch check ")
 
+	resp, ok := trigger.CheckEmptyOrNilExternalConfig(config)
```

Comment:
Why are we checking all the config for certificate?. There can be the many cases where all external config are nil but not certificates. 

if config.ExternalOS == nil {
		return ExternalOSPGNillResp(config), true
	}

	if config.ExternalPG == nil {
		return ExternalOSPGNillResp(config), true
	}

	if IsEmptyExternalOS(config.ExternalOS) {
		return ExternalOSPGEmptyResp(config), true
	}

	if IsEmptyExternalPG(config.ExternalPG) {
		return ExternalOSPGEmptyResp(config), true
	}